src/pip/_internal/commands/debug.py

```diff
@@ -118,7 +119,7 @@ def show_tags(options: Values) -> None:
 
     if options.verbose < 1 and len(tags) > tag_limit:
         tags_limited = True
-        tags = tags[:tag_limit]
+        tags = list(tags)[:tag_limit]
```

Comment:
This would make the list unpredictable and useless. The ordering (before this PR) is from restrictive to general and quite informative.

Reply:
Hm, maybe I can rename the current `get_tags` method to `get_sorted_tags` and change all callers where order seems important?

Reply:
Or we can add an `iter_tags` that creates an iterator for the tags, and call that in `debug`. `get_tags` would also call it and cache the result in a set.

Reply:
Thanks, I implemented this, except in what I pushed `iter_tags` is named `get_sorted_tags` (seems more explicit that order matters + return type was already a list)